hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionRegistryURIFactory.java

```diff
@@ -39,4 +39,10 @@ public interface ConnectionRegistryURIFactory {
    * {@link ConnectionRegistryFactory}.
    */
   String getScheme();
+
+  /**
+   * Validate the given {@code uri}.
+   * @throws IOException if this is not a valid connection registry URI.
+   */
+  void validate(URI uri) throws IOException;
```

Comment:
Maybe this interface should be `boolean isValid(URI)`, and then the caller is free to throw or not.

Reply:
Throwing exception could let the upper layer know the details about why this is not valid.

Reply:
Yeah, I understand. This enforces try-catch flow control where if/else would be better. For this kind of thing, I really like the `Result` based API that is catching on in other languages.

Anyway, maybe it should throw something besides `IOException` ? IllegatStateException, for example?

Reply:
Maybe something like UnsupportedProtocol or IllegalArgument? But IllegalArgument is a RuntimeException, developers may miss to catch it and cause some fatal errors...

Reply:
I think a RuntimeException is okay. It happens often enough in the JDK libraries, where they will also add a note about it in the javadoc.

Reply:
Checked the implementation, now we have these errors

No protocol scheme
No factory registered for the scheme

For zk based registry
Empty zk server string, i.e, empty uri authority
Empty zk parent path, i.e, empty uri path

For rpc based registry
Empty bootstrap nodes, i.e, empty uri authority

In general, there are no accurate exception types for these errors, and since we may add new checks for different registry implementations in the future, I prefer we still keep the `throws IOException` declaration, and can file new issues to add some specific exceptions which extend HBaseIOException for these cases.